### PR TITLE
refactor/core-to-review -> refactor/core

### PR DESCRIPTION
**Pull request by InuDial**
## Summary

- Problem:
- Why it matters:
- What changed:
- What did not change (scope boundary):

## Linked Issues

- Closes #
- Related #

## Change Type

- [ ] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Security hardening
- [ ] CI / workflow / release

## Touched Areas

- [ ] Kernel / policy / approvals
- [ ] Contracts / protocol / spec
- [ ] Daemon / CLI / install
- [ ] Providers / routing
- [ ] Tools
- [ ] Browser automation
- [ ] Channels / integrations
- [ ] ACP / conversation / session runtime
- [ ] Memory / context assembly
- [ ] Config / migration / onboarding
- [ ] Docs / contributor workflow
- [ ] CI / release / workflows

## Risk Track

- [ ] Track A (routine / low-risk)
- [ ] Track B (higher-risk / policy-impacting)

If Track B, fill these in:

- Risk notes:
- Rollout / guardrails:
- Rollback path:

## Validation

- [ ] `cargo fmt --all -- --check`
- [ ] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [ ] Rust tests match the touched packages / feature surface (for example `cargo test --workspace --locked`, `cargo test --workspace --all-features --locked`, `task verify`, `task verify:changed`, or equivalent exact commands documented below)
- [ ] If full workspace / all-features Rust tests were not run locally, explain why and cite CI evidence or a known baseline blocker
- [ ] Relevant architecture / dep-graph / docs checks for touched areas
- [ ] Additional scenario, benchmark, or manual checks when behavior changed
- [ ] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [ ] If tests mutate process-global env: document how state is restored or serialized

Commands and evidence:

```text
Paste the exact commands you ran and summarize the result.
```

## User-visible / Operator-visible Changes

- None, or describe the exact change:

## Failure Recovery

- Fast rollback or disable path:
- Observable failure symptoms reviewers should watch for:

## Reviewer Focus

- Point reviewers at the files, edge cases, or risk seams that deserve the most attention.

### CODE REVIEW

File: crates/tools/src/file/error.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#[non_exhaustive]
+pub enum FileToolError {
+    #[error("{0}")]
+    Access(
+        #[from]
+        #[source]
+        FsAccessError,
+    ),
+    #[error("failed to decode {path} as UTF-8: {source}", path = .path.display())]
+    InvalidUtf8 {
```
Comment (InuDial) [inline crates/tools/src/file/error.rs]: Maybe other encoding methods needed?

File: crates/loong-core/src/policy/tests.rs
Comment (InuDial) [inline crates/loong-core/src/policy/tests.rs:1]: This should be test_support.rs

File: crates/contracts/src/action_grant.rs [hunk 2]
```diff
@@ -1,33 +1,143 @@
...
+impl PolicyId {
+    #[must_use]
+    pub const fn new(value: u64) -> Self {
+        Self(value)
+    }
+}
+
+/// Source location where a policy entered its runtime pipeline.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PolicyRegistrationSource {
```
Comment (InuDial) [inline crates/contracts/src/action_grant.rs:63]: May be more graceful?

File: crates/tools/src/file/edit.rs [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EditOutput {
+    pub tool_name: String,
+    pub path: PathBuf,
+    pub before: String,
+    pub after: String,
+    pub replacements_made: usize,
+    pub edit_blocks_applied: usize,
+}
```
Comment (InuDial) [inline crates/tools/src/file/edit.rs:47]: Redundant?

Comment (InuDial) [reply crates/tools/src/file/edit.rs:47]: Possibly different versions of this tool is needed.
